internal/gh/rest.go

```diff
@@ -0,0 +1,219 @@
+        }
+        
+    }
+
+    req.URL.RawQuery = query.Encode()
+
+    return nil
+}
+
+//  Entry point for the Force Push processing, setting the url, query paramenters and creating the client.
+func GetForcePushInfo(owner, repo, token, branch string) (int, error){
+
+    slog.Default().Info("Getting repo info - getForcePushInfo method")
+    repoActivityUrl := baseURL + "/repos/" + owner + "/" + repo + "/activity"
+
+    queryParameters := map[string]string{
+        "per_page" : "100",
+        "activity_type" : "force_push",
+        "ref" : branch,
+    }
+
+    headerParameters := map[string]string{
+        "Content-Type" : "application/json",
+        "Authorization": "Bearer "+token,
+    }
+
+    client := &http.Client{}
+
+    NumberForcePush, err := processForcePushRequest(client, repoActivityUrl, queryParameters, headerParameters) 
+    if err != nil {
+        slog.Default().Error("Getting repo info falied - %v getForcePushInfo method", err)
+        return 0, err
+    }    
+    return NumberForcePush, nil
+
+}
+
+//  Execute the HTTP request and return response
+func executeHTTPRequest(client *http.Client, req *http.Request) (*http.Response, error){
+
+    resp, err := client.Do(req)
+	RequestCounter++
+	if err != nil {
+        return nil, err
+	}
+    return resp, nil
+
+}
+
+//  Handle calls to other methods (createHTTPRequest, executeHTTPRequest, processHttpResponse) is handled and loop is added so that requests are processed till there no rel = "next"
+func processForcePushRequest(client *http.Client, repoActivityUrl string, queryParameters, headerParameters map[string] string) (int, error) {
+
+    NumberForcePush := 0
+    hasNext := true
+
+    // Create an initial HTTP Request and set header and query parameters
+    httpReq, err := createHttpRequest(client, repoActivityUrl,"GET", nil, queryParameters, headerParameters)
+    if err != nil {
+        slog.Default().Error("Failed to create HTTP request: %v", err)
+        return  0, err
+    }
+
+    for hasNext{
+
+        // Execute the HTTP Request
+        resp, err := executeHTTPRequest(client, httpReq)
+        if err != nil {
+		    slog.Default().Error("Failed to execute HTML request: %v - processForcePushRequest method", err)
+		    return 0, err
+	    }
+
+        defer resp.Body.Close()
+        
+        // Process the response as result data type
+        res, err := processHttpResponse(resp)
+        if err != nil {
+	        slog.Default().Error("Failed to process HTTP response: %v - processForcePushRequest method", err)
+	        return 0, err
+	    }
+
+        NumberForcePush += len(res)
+
+        // Check if Next Page exists         
+        hasNext, err = checkIfNextPageExists(resp)
+        if err != nil {
+	        slog.Default().Error("Failed to check for next pages in force pushes: %v - processHttpResponse method", err)
+	        return 0, err
+	    }        
+        // If next page exists get the url of the next page
+        if hasNext {
```

Comment:
For the go code to be more idiomatic you usually put the main control flow outside of if statements. In this case you would do it like this:

```suggestion
        if !hasNext {
            break // exit loop as there are no more pages
         }

        // if we get here a new page exists so we can call getNextURL()
        repoActivityUrl, err = getNextURL(resp)
        // ... 
```